lib/segment/src/segment.rs

```diff
@@ -824,11 +824,17 @@ impl SegmentEntry for Segment {
                     segment.payload_index.borrow_mut().drop(internal_id)?;
                     segment.id_tracker.borrow_mut().drop(point_id)?;
 
-                    // Propagate point deletion to all its vectors
-                    for vector_data in segment.vector_data.values() {
-                        let mut vector_storage = vector_data.vector_storage.borrow_mut();
-                        vector_storage.delete_vector(internal_id)?;
-                    }
+                    // Before, we propagated point deletions to also delete its vectors. This turns
```

Comment:
Why is it OK to stop suddenly deleting vectors regarding correctness and accumulation over time?

Reply:
The point is still marked as deleted. If we have many deleted points, our vacuum optimizer will reoptimize the segment. That will drop the points (and thus these vectors).

Reply:
Note that these deletions were propagated so the vacuum optimizer _might_ kick in a bit earlier to clean-up many deleted vectors, as it would count deleted points AND deleted vectors.

Now we just optimize this situation if vectors are explicitly deleted, and we still have the vacuum optimizer to catch many deleted points (which includes vectors).

Sorry if this all is a bit confusing.